### PR TITLE
docs: Add API docs for Admin Partitions

### PR DESCRIPTION
**Pull request by kisunji**

### CODE REVIEW

File: website/content/api-docs/admin-partitions.mdx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+[blocking queries](/api/features/blocking),
+[consistency modes](/api/features/consistency),
+[agent caching](/api/features/caching), and
+[required ACLs](/api#authentication).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required                          |
+| ---------------- | ----------------- | ------------- | ------------------------------------- |
+| `NO`             | `consistent`      | `none`        | `operator:read` or `none`<sup>1</sup> |
+
+<sup>1</sup> A non-anonymous token in a non-default partition can read its own partition.
```
Comment (kisunji) [inline website/content/api-docs/admin-partitions.mdx:86]: Would like some eyes here

File: website/content/api-docs/admin-partitions.mdx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+The table below shows this endpoint's support for
+[blocking queries](/api/features/blocking),
+[consistency modes](/api/features/consistency),
+[agent caching](/api/features/caching), and
+[required ACLs](/api#authentication).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required    |
+| ---------------- | ----------------- | ------------- | --------------- |
+| `NO`             | `consistent`      | `none`        | `operator:read` |
```
Comment (kisunji) [inline website/content/api-docs/admin-partitions.mdx:232]: Would like eyes here

File: website/content/api-docs/admin-partitions.mdx (hunk 1 above)
Comment (kyhavlov) [reply website/content/api-docs/admin-partitions.mdx:86]: For the consistency modes, stale queries are supported as well (so it should be `all`).

The token also doesn't have to be non-default, just non-anonymous.

File: website/content/api-docs/admin-partitions.mdx (hunk 2 above)
Comment (kyhavlov) [reply website/content/api-docs/admin-partitions.mdx:232]: The consistency mode should be `all`, like with the read endpoint above

**Review by kyhavlov — COMMENTED** [review]
Looking good, just a couple comments

File: website/content/api-docs/admin-partitions.mdx [hunk 3]
```diff
@@ -0,0 +1,258 @@
+---
+layout: api
```
Comment (rboyer) [inline website/content/api-docs/admin-partitions.mdx:2]: Isn't there a sidebar change required to get this to be linked?

Comment (rboyer) [reply website/content/api-docs/admin-partitions.mdx:2]: From the failing vercel preview:
```
> Build error occurred
14:42:19.870 | Error:
14:42:19.871 | Error: Unlinked pages found in the content/api-docs directory.
14:42:19.871 |  
14:42:19.871 | Please add these paths to the "content/api-docs" nav data file, or remove the .mdx files. If you want the pages to exist but not be linked in the navigation, you can add a "hidden" property to the page object in the navigation structure.
14:42:19.871 |  
14:42:19.872 | [
14:42:19.872 | "admin-partitions"
14:42:19.872 | ]
14:42:19.872

<br class="Apple-interchange-newline">
```

File: website/content/api-docs/admin-partitions.mdx (hunk 1 above)
Comment (dnephin) [reply website/content/api-docs/admin-partitions.mdx]: Are stale queries supported? I saw this TODO in the code: https://github.com/hashicorp/consul/blob/main/proto/pbcommon/common.go#L148-L152

And I don't see a field on `pbcommon.ReadRequest` to set AllowStale.

Comment (kyhavlov) [reply website/content/api-docs/admin-partitions.mdx]: Oops, you're right, we have the plumbing supported for this in `ForwardGRPC` and `canServeReadRequest`, but I didn't realize `pbcommon.ReadRequest` was never updated for it. Sorry for the confusion @kisunji!

**Review by rboyer — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/528747.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 729ff425ff992207ea2d235390951fa76ad67da8 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 729ff425ff992207ea2d235390951fa76ad67da8 onto `release/1.11.x` succeeded!

**Pull request merged**